src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/validate/PNCIdentifier.kt

```diff
@@ -0,0 +1,64 @@
+package uk.gov.justice.digital.hmpps.personrecord.validate
+
+import org.slf4j.LoggerFactory
+import java.time.LocalDate
+
+const val LONG_PNC_ID_LENGTH = 10
+
+data class PNCIdentifier(private val inputPncId: String? = null) {
+  val pncId: String?
+    get() = toCanonicalForm(inputPncId)
+
+  fun isEquivalentTo(otherPncId: PNCIdentifier?): Boolean {
```

Comment:
I wonder if this could be the `equals` method now? Not sure it is a good idea but worth thinking about